### ISSUE TITLE
Modal closing when selecting datepicker

### ISSUE THREAD

**Issue by vincentwongso** [Semantic-Org/Semantic-UI-React#2945]
## Bug Report
I am using [react-datepicker](https://github.com/Hacker0x01/react-datepicker) for the following test.
When selecting a date in the datepicker inside a modal, the modal also close together with the datepicker. It's working fine with react-bootstrap and react-material modal, so I assume the problem is with semantic ui modal.

### Steps
1. Create a modal using semantic ui.
2. Add datepicker inside modal.
3. Select the date and when you click the date, it will close the datepicker as well as the modal.

Please see codesandbox below to demonstrate the issue.

### Expected Result
When clicking the datepicker it should only close the datepicker.

### Actual Result
When clicking the datepicker it close the datepicker as well as the modal.

### Version
React semantic ui: 0.81.1
react-datepicker: 1.5.0

### Testcase
https://codesandbox.io/s/y7q96n58q9

**Comment by honzajerabek** [comment]
Hey, I'm using the same setup. This happened to me as well, when the `react-datepicker` was overlapping the modal's edge and I clicked the date which was positioned out of the modal.. Technically, I clicked out of the modal -> modal closes

Solved temporarily with not passing `onClose` prop and handling close myself

edit: I see your test case does just that..

**Comment by vincentwongso** [comment]
@honzajerabek thank you so much for the info, do you have any example on your workaround. 
I tried passing onClose callback but the event that get passed to that callback function is MouseEvent and the data is the modal itself.

**Comment by herbertpimentel** [comment]
This only happening when using the property `trigger` to open the modal, if you use the property `open` it does not happen.

**Comment by vincentwongso** [comment]
It's happening when I use `open` as well. Even when the I handle the `close` function manually, it's still happening.
Please see the following example:
https://codesandbox.io/s/z6xoo25p3p

**Comment by drew23c** [comment]
I added showModal to handleChangeDate so that it closes when the user changes the date:

https://codesandbox.io/s/olopwjk9q6

I edited the link

**Comment by herbertpimentel** [comment]
It closes because your event `handleChangeDate` changes `showModal` to false. 

```javascript
handleChangeDate = (date) => {
    this.setState({
      date,
      showModal: false
    });
  };
```

**Comment by drew23c** [comment]
Ok. I think I understand now. Thank you!

**Comment by vincentwongso** [comment]
This issue has been fixed with the release of 0.82.1, the fix is on #3015. So I think we can close this ticket.
Thanks @layershifter .
